### ISSUE TITLE
Unable to `pip install onnx` on Ubuntu 22.04

### ISSUE THREAD

**Issue by kkloberdanz** [onnx/onnx#4179]
# Bug Report

### Is the issue related to model conversion?
No

### Describe the bug
Unable to `pip install onnx` on Ubuntu 22.04

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*): Linux Ubuntu 22.04
- ONNX version (*e.g. 1.7*):  1.11.0
- Python version: Python 3.10.4
- GCC/Compiler version (if compiling from source): gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0
- CMake version: cmake version 3.22.1
- Protobuf version: libprotoc 3.12.4
- Visual Studio version (if applicable): N/A

### Reproduction instructions
- Describe the code to reproduce the behavior.

Create a `Dockerfile` with this content:
```
FROM ubuntu:22.04

RUN apt update && apt install -y cmake python3 python3-pip python3-virtualenv

RUN virtualenv venv -p `which python3`

RUN . venv/bin/activate && pip install onnx
```

Then build the `Dockerfile` like so:
```bash
docker build . -t onnx-ubuntu
```

Attempting to install ONNX on Ubuntu 22.04 results in the error message found below in the Notes section.

- Attach the ONNX model to the issue (where applicable)
Bug is when installing onnx, so N/A

### Expected behavior
The expected behavior is for the installation to not fail and for the package to be installed

### Notes

Any additional information
```
# pip install onnx
Collecting onnx
  Using cached onnx-1.11.0.tar.gz (9.9 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy>=1.16.6 in /venv/lib/python3.10/site-packages (from onnx) (1.22.3)
Requirement already satisfied: protobuf>=3.12.2 in /venv/lib/python3.10/site-packages (from onnx) (3.20.1)
Requirement already satisfied: typing-extensions>=3.6.2.1 in /venv/lib/python3.10/site-packages (from onnx) (4.2.0)
Building wheels for collected packages: onnx
  Building wheel for onnx (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [224 lines of output]
      /venv/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running bdist_wheel
      running build
      running build_py
      running create_version
      running cmake_build
      Using cmake args: ['/usr/bin/cmake', '-DPYTHON_INCLUDE_DIR=/usr/include/python3.10', '-DPYTHON_EXECUTABLE=/venv/bin/python', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cpython-310-x86_64-linux-gnu.so', '-DCMAKE_BUILD_TYPE=Release', '-DONNX_ML=1', '/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f']
      -- The C compiler identification is GNU 11.2.0
      -- The CXX compiler identification is GNU 11.2.0
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found PythonInterp: /venv/bin/python (found version "3.10.4")
      -- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.10.so (found version "3.10.4")
      -- Found Protobuf: /usr/lib/x86_64-linux-gnu/libprotobuf.a (found version "3.12.4")
      Generated: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-ml.proto
      Generated: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-operators-ml.proto
      Generated: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-data.proto
      CMake Warning at CMakeLists.txt:451 (find_package):
        By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
        asked CMake to find a package configuration file provided by "pybind11",
        but CMake did not find one.
      
        Could not find a package configuration file provided by "pybind11"
        (requested version 2.2) with any of the following names:
      
          pybind11Config.cmake
          pybind11-config.cmake
      
        Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
        "pybind11_DIR" to a directory containing one of the above files.  If
        "pybind11" provides a separate development package or SDK, be sure it has
        been installed.
      
      
      --
      -- ******** Summary ********
      --   CMake version             : 3.22.1
      --   CMake command             : /usr/bin/cmake
      --   System                    : Linux
      --   C++ compiler              : /usr/bin/c++
      --   C++ compiler version      : 11.2.0
      --   CXX flags                 :  -Wnon-virtual-dtor
      --   Build type                : Release
      --   Compile definitions       : __STDC_FORMAT_MACROS
      --   CMAKE_PREFIX_PATH         :
      --   CMAKE_INSTALL_PREFIX      : /usr/local
      --   CMAKE_MODULE_PATH         :
      --
      --   ONNX version              : 1.11.0
      --   ONNX NAMESPACE            : onnx
      --   ONNX_USE_LITE_PROTO       : OFF
      --   USE_PROTOBUF_SHARED_LIBS  : OFF
      --   Protobuf_USE_STATIC_LIBS  : ON
      --   ONNX_DISABLE_EXCEPTIONS   : OFF
      --   ONNX_WERROR               : OFF
      --   ONNX_BUILD_TESTS          : OFF
      --   ONNX_BUILD_BENCHMARKS     : OFF
      --   ONNXIFI_DUMMY_BACKEND     : OFF
      --   ONNXIFI_ENABLE_EXT        : OFF
      --
      --   Protobuf compiler         : /usr/bin/protoc
      --   Protobuf includes         : /usr/include
      --   Protobuf libraries        : /usr/lib/x86_64-linux-gnu/libprotobuf.a
      --   BUILD_ONNX_PYTHON         : ON
      --     Python version        :
      --     Python executable     : /venv/bin/python
      --     Python includes       : /usr/include/python3.10
      -- Configuring done
      -- Generating done
      -- Build files have been written to: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build
      [  4%] Building C object CMakeFiles/onnxifi_loader.dir/onnx/onnxifi_loader.c.o
      [  4%] Running gen_proto.py on onnx/onnx.in.proto
      [  4%] Building C object CMakeFiles/onnxifi_dummy.dir/onnx/onnxifi_dummy.c.o
      Processing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/onnx.in.proto
      Writing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-ml.proto
      Writing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-ml.proto3
      generating /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx_pb.py
      [  5%] Linking C static library libonnxifi_loader.a
      [  7%] Running C++ protocol buffer compiler on /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-ml.proto
      [  8%] Linking C shared library libonnxifi_dummy.so
      [  8%] Built target onnxifi_loader
      [  9%] Building C object CMakeFiles/onnxifi_wrapper.dir/onnx/onnxifi_wrapper.c.o
      [  9%] Built target onnxifi_dummy
      Writing mypy to onnx/onnx_ml_pb2.pyi
      [  9%] Built target gen_onnx_proto
      [ 12%] Running gen_proto.py on onnx/onnx-operators.in.proto
      [ 12%] Running gen_proto.py on onnx/onnx-data.in.proto
      Processing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/onnx-operators.in.proto
      Writing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-operators-ml.proto
      Writing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-operators-ml.proto3
      generating /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx_operators_pb.py
      Processing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/onnx-data.in.proto
      Writing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-data.proto
      Writing /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-data.proto3
      generating /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx_data_pb.py
      [ 15%] Running C++ protocol buffer compiler on /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-operators-ml.proto
      [ 15%] Running C++ protocol buffer compiler on /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-data.proto
      [ 16%] Linking C shared module libonnxifi.so
      [ 16%] Built target onnxifi_wrapper
      Writing mypy to onnx/onnx_operators_ml_pb2.pyi
      Writing mypy to onnx/onnx_data_pb2.pyi
      [ 16%] Built target gen_onnx_operators_proto
      [ 16%] Built target gen_onnx_data_proto
      [ 19%] Building CXX object CMakeFiles/onnx_proto.dir/onnx/onnx-operators-ml.pb.cc.o
      [ 19%] Building CXX object CMakeFiles/onnx_proto.dir/onnx/onnx-ml.pb.cc.o
      [ 21%] Building CXX object CMakeFiles/onnx_proto.dir/onnx/onnx-data.pb.cc.o
      [ 22%] Linking CXX static library libonnx_proto.a
      [ 30%] Built target onnx_proto
      [ 36%] Building CXX object CMakeFiles/onnx.dir/onnx/common/path.cc.o
      [ 36%] Building CXX object CMakeFiles/onnx.dir/onnx/common/assertions.cc.o
      [ 36%] Building CXX object CMakeFiles/onnx.dir/onnx/common/interned_strings.cc.o
      [ 38%] Building CXX object CMakeFiles/onnx.dir/onnx/common/status.cc.o
      [ 39%] Building CXX object CMakeFiles/onnx.dir/onnx/common/model_helpers.cc.o
      [ 39%] Building CXX object CMakeFiles/onnx.dir/onnx/checker.cc.o
      [ 40%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/attr_proto_util.cc.o
      [ 42%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/controlflow/defs.cc.o
      [ 45%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/controlflow/old.cc.o
      [ 46%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/data_type_utils.cc.o
      [ 46%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/function.cc.o
      [ 47%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/generator/defs.cc.o
      [ 49%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/generator/old.cc.o
      [ 50%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/logical/defs.cc.o
      [ 52%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/logical/old.cc.o
      [ 53%] Building CXX object CMakeFiles/onnx.dir/onnx/common/ir_pb_converter.cc.o
      [ 54%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/math/defs.cc.o
      [ 56%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/math/old.cc.o
      [ 57%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/nn/defs.cc.o
      [ 59%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/nn/old.cc.o
      [ 60%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/object_detection/defs.cc.o
      [ 61%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/object_detection/old.cc.o
      [ 63%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/optional/defs.cc.o
      [ 64%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/parser.cc.o
      [ 66%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/printer.cc.o
      [ 67%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/quantization/defs.cc.o
      [ 69%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/quantization/old.cc.o
      [ 70%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/reduction/defs.cc.o
      In file included from /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/common/ir_pb_converter.h:10,
                       from /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/common/ir_pb_converter.cc:8:
      In constructor ‘onnx::Dimension::Dimension(onnx::Dimension&&)’,
          inlined from ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = onnx::Dimension; _Args = {onnx::Dimension}; _Tp = onnx::Dimension]’ at /usr/include/c++/11/ext/new_allocator.h:162:4,
          inlined from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = onnx::Dimension; _Args = {onnx::Dimension}; _Tp = onnx::Dimension]’ at /usr/include/c++/11/bits/alloc_traits.h:516:17,
          inlined from ‘void std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {onnx::Dimension}; _Tp = onnx::Dimension; _Alloc = std::allocator<onnx::Dimension>]’ at /usr/include/c++/11/bits/vector.tcc:115:30,
          inlined from ‘void std::vector<_Tp, _Alloc>::push_back(std::vector<_Tp, _Alloc>::value_type&&) [with _Tp = onnx::Dimension; _Alloc = std::allocator<onnx::Dimension>]’ at /usr/include/c++/11/bits/stl_vector.h:1204:21,
          inlined from ‘std::vector<onnx::Dimension> onnx::tensorShapeProtoToDimensions(const onnx::TensorShapeProto&)’ at /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/common/ir_pb_converter.cc:201:21:
      /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/common/ir.h:74:8: warning: ‘<unnamed>.onnx::Dimension::dim’ may be used uninitialized [-Wmaybe-uninitialized]
         74 | struct Dimension final {
            |        ^~~~~~~~~
      /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/common/ir_pb_converter.cc: In function ‘std::vector<onnx::Dimension> onnx::tensorShapeProtoToDimensions(const onnx::TensorShapeProto&)’:
      /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/onnx/common/ir_pb_converter.cc:201:32: note: ‘<anonymous>’ declared here
        201 |       dims.push_back(Dimension());
            |                                ^
      [ 71%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/reduction/old.cc.o
      [ 73%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/rnn/defs.cc.o
      [ 74%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/rnn/old.cc.o
      [ 76%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/schema.cc.o
      [ 77%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/sequence/defs.cc.o
      [ 78%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/shape_inference.cc.o
      [ 80%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/tensor/defs.cc.o
      [ 81%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/tensor/old.cc.o
      [ 83%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/tensor/utils.cc.o
      [ 84%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/tensor_proto_util.cc.o
      [ 85%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/tensor_util.cc.o
      [ 87%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/traditionalml/defs.cc.o
      [ 88%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/traditionalml/old.cc.o
      [ 90%] Building CXX object CMakeFiles/onnx.dir/onnx/defs/training/defs.cc.o
      [ 91%] Building CXX object CMakeFiles/onnx.dir/onnx/onnxifi_utils.cc.o
      [ 92%] Building CXX object CMakeFiles/onnx.dir/onnx/shape_inference/implementation.cc.o
      [ 94%] Building CXX object CMakeFiles/onnx.dir/onnx/version_converter/convert.cc.o
      [ 95%] Building CXX object CMakeFiles/onnx.dir/onnx/version_converter/helper.cc.o
      [ 97%] Linking CXX static library libonnx.a
      [ 97%] Built target onnx
      [ 98%] Building CXX object CMakeFiles/onnx_cpp2py_export.dir/onnx/cpp2py_export.cc.o
      [100%] Linking CXX shared module onnx_cpp2py_export.cpython-310-x86_64-linux-gnu.so
      /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libprotobuf.a(arena.o): relocation R_X86_64_TPOFF32 against hidden symbol `_ZN6google8protobuf8internal9ArenaImpl13thread_cache_E' can not be used when making a shared object
      /usr/bin/ld: failed to set dynamic section sizes: bad value
      collect2: error: ld returned 1 exit status
      gmake[2]: *** [CMakeFiles/onnx_cpp2py_export.dir/build.make:101: onnx_cpp2py_export.cpython-310-x86_64-linux-gnu.so] Error 1
      gmake[1]: *** [CMakeFiles/Makefile2:229: CMakeFiles/onnx_cpp2py_export.dir/all] Error 2
      gmake: *** [Makefile:136: all] Error 2
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/setup.py", line 336, in <module>
          setuptools.setup(
        File "/venv/lib/python3.10/site-packages/setuptools/__init__.py", line 153, in setup
          return distutils.core.setup(**attrs)
        File "/usr/lib/python3.10/distutils/core.py", line 148, in setup
          dist.run_commands()
        File "/usr/lib/python3.10/distutils/dist.py", line 966, in run_commands
          self.run_command(cmd)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/venv/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 299, in run
          self.run_command('build')
        File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/usr/lib/python3.10/distutils/command/build.py", line 135, in run
          self.run_command(cmd_name)
        File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/setup.py", line 232, in run
          self.run_command('cmake_build')
        File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/setup.py", line 226, in run
          subprocess.check_call(build_args)
        File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['/usr/bin/cmake', '--build', '.', '--', '-j', '16']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for onnx
  Running setup.py clean for onnx
Failed to build onnx
Installing collected packages: onnx
  Running setup.py install for onnx ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for onnx did not run successfully.
  │ exit code: 1
  ╰─> [125 lines of output]
      /venv/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running install
      /venv/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      running create_version
      running cmake_build
      Using cmake args: ['/usr/bin/cmake', '-DPYTHON_INCLUDE_DIR=/usr/include/python3.10', '-DPYTHON_EXECUTABLE=/venv/bin/python', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cpython-310-x86_64-linux-gnu.so', '-DCMAKE_BUILD_TYPE=Release', '-DONNX_ML=1', '/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f']
      Generated: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-ml.proto
      Generated: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-operators-ml.proto
      Generated: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build/onnx/onnx-data.proto
      CMake Warning at CMakeLists.txt:451 (find_package):
        By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
        asked CMake to find a package configuration file provided by "pybind11",
        but CMake did not find one.
      
        Could not find a package configuration file provided by "pybind11"
        (requested version 2.2) with any of the following names:
      
          pybind11Config.cmake
          pybind11-config.cmake
      
        Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
        "pybind11_DIR" to a directory containing one of the above files.  If
        "pybind11" provides a separate development package or SDK, be sure it has
        been installed.
      
      
      --
      -- ******** Summary ********
      --   CMake version             : 3.22.1
      --   CMake command             : /usr/bin/cmake
      --   System                    : Linux
      --   C++ compiler              : /usr/bin/c++
      --   C++ compiler version      : 11.2.0
      --   CXX flags                 :  -Wnon-virtual-dtor
      --   Build type                : Release
      --   Compile definitions       : __STDC_FORMAT_MACROS
      --   CMAKE_PREFIX_PATH         :
      --   CMAKE_INSTALL_PREFIX      : /usr/local
      --   CMAKE_MODULE_PATH         :
      --
      --   ONNX version              : 1.11.0
      --   ONNX NAMESPACE            : onnx
      --   ONNX_USE_LITE_PROTO       : OFF
      --   USE_PROTOBUF_SHARED_LIBS  : OFF
      --   Protobuf_USE_STATIC_LIBS  : ON
      --   ONNX_DISABLE_EXCEPTIONS   : OFF
      --   ONNX_WERROR               : OFF
      --   ONNX_BUILD_TESTS          : OFF
      --   ONNX_BUILD_BENCHMARKS     : OFF
      --   ONNXIFI_DUMMY_BACKEND     : OFF
      --   ONNXIFI_ENABLE_EXT        : OFF
      --
      --   Protobuf compiler         : /usr/bin/protoc
      --   Protobuf includes         : /usr/include
      --   Protobuf libraries        : /usr/lib/x86_64-linux-gnu/libprotobuf.a
      --   BUILD_ONNX_PYTHON         : ON
      --     Python version        :
      --     Python executable     : /venv/bin/python
      --     Python includes       : /usr/include/python3.10
      -- Configuring done
      -- Generating done
      -- Build files have been written to: /tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/.setuptools-cmake-build
      Consolidate compiler generated dependencies of target onnxifi_loader
      Consolidate compiler generated dependencies of target onnxifi_dummy
      [  2%] Built target gen_onnx_proto
      [  7%] Built target onnxifi_dummy
      [  8%] Built target onnxifi_loader
      Consolidate compiler generated dependencies of target onnxifi_wrapper
      [ 12%] Built target gen_onnx_operators_proto
      [ 14%] Built target gen_onnx_data_proto
      [ 16%] Built target onnxifi_wrapper
      Consolidate compiler generated dependencies of target onnx_proto
      [ 30%] Built target onnx_proto
      Consolidate compiler generated dependencies of target onnx
      [ 97%] Built target onnx
      Consolidate compiler generated dependencies of target onnx_cpp2py_export
      [ 98%] Linking CXX shared module onnx_cpp2py_export.cpython-310-x86_64-linux-gnu.so
      /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libprotobuf.a(arena.o): relocation R_X86_64_TPOFF32 against hidden symbol `_ZN6google8protobuf8internal9ArenaImpl13thread_cache_E' can not be used when making a shared object
      /usr/bin/ld: failed to set dynamic section sizes: bad value
      collect2: error: ld returned 1 exit status
      gmake[2]: *** [CMakeFiles/onnx_cpp2py_export.dir/build.make:101: onnx_cpp2py_export.cpython-310-x86_64-linux-gnu.so] Error 1
      gmake[1]: *** [CMakeFiles/Makefile2:229: CMakeFiles/onnx_cpp2py_export.dir/all] Error 2
      gmake: *** [Makefile:136: all] Error 2
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/setup.py", line 336, in <module>
          setuptools.setup(
        File "/venv/lib/python3.10/site-packages/setuptools/__init__.py", line 153, in setup
          return distutils.core.setup(**attrs)
        File "/usr/lib/python3.10/distutils/core.py", line 148, in setup
          dist.run_commands()
        File "/usr/lib/python3.10/distutils/dist.py", line 966, in run_commands
          self.run_command(cmd)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/venv/lib/python3.10/site-packages/setuptools/command/install.py", line 68, in run
          return orig.install.run(self)
        File "/usr/lib/python3.10/distutils/command/install.py", line 619, in run
          self.run_command('build')
        File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/usr/lib/python3.10/distutils/command/build.py", line 135, in run
          self.run_command(cmd_name)
        File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/setup.py", line 232, in run
          self.run_command('cmake_build')
        File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
          cmd_obj.run()
        File "/tmp/pip-install-sat41brg/onnx_862fcf0345b34d2a95421c32a2fee89f/setup.py", line 226, in run
          subprocess.check_call(build_args)
        File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['/usr/bin/cmake', '--build', '.', '--', '-j', '16']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> onnx

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Comment by jcwchen** [comment]
Hi @kkloberdanz,
Same as https://github.com/onnx/onnx/issues/4146. Please note that official ONNX hasn't released a Python 3.10 wheel yet so actually you are building ONNX from source. To build ONNX from source, you need to set config `export CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"` with shared protobuf installed or install static Protobuf from source. You can check the instruction here: https://github.com/onnx/onnx#build-onnx-from-source.

**Comment by kkloberdanz** [comment]
I can confirm that this works:
```
(venv) root@01fb5114864e:/# export CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"
(venv) root@01fb5114864e:/# pip install onnx
Collecting onnx
  Using cached onnx-1.11.0.tar.gz (9.9 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy>=1.16.6 in /venv/lib/python3.10/site-packages (from onnx) (1.22.3)
Requirement already satisfied: protobuf>=3.12.2 in /venv/lib/python3.10/site-packages (from onnx) (3.20.1)
Requirement already satisfied: typing-extensions>=3.6.2.1 in /venv/lib/python3.10/site-packages (from onnx) (4.2.0)
Building wheels for collected packages: onnx
  Building wheel for onnx (setup.py) ... done
  Created wheel for onnx: filename=onnx-1.11.0-cp310-cp310-linux_x86_64.whl size=11745705 sha256=19c43be087322b5eb90d8558fa8d53467a502e246c1927783015d1840819c226
  Stored in directory: /root/.cache/pip/wheels/06/fe/59/bfb3b042234d9578b518ddc0a86bd8f9c4a8d60ebf4da61375
Successfully built onnx
Installing collected packages: onnx
Successfully installed onnx-1.11.0
```

**Comment by jcwchen** [comment]
Thanks for the quick trial! Then, I will close this issue. Feel free to reopen it if you still have other ONNX questions.

**Comment by pramenku** [comment]
> Hi @kkloberdanz, Same as #4146. Please note that official ONNX hasn't released a Python 3.10 wheel yet so actually you are building ONNX from source. To build ONNX from source, you need to set config `export CMAKE_ARGS="-DONNX_USE_PROTOBUF_SHARED_LIBS=ON"` with shared protobuf installed or install static Protobuf from source. You can check the instruction here: https://github.com/onnx/onnx#build-onnx-from-source.

Thanks. Perfect reason and solution. It helps. :)
